### PR TITLE
Update OTEL exporters to use new API

### PR DESCRIPTION
**Pull request by pavolloffay**
~Currently blocked by https://github.com/open-telemetry/opentelemetry-collector/issues/872 as we need a new translator from `pdata.Traces` to Jaeger data model.~

### CODE REVIEW

**Comment by pavolloffay** [comment]
@objectiser this is ready for review

File: cmd/opentelemetry-collector/app/exporter/kafka/factory.go [hunk 1]
```diff
@@ -42,13 +42,15 @@ type Factory struct {
...
+
 // Type gets the type of exporter.
-func (Factory) Type() string {
+func (Factory) Type() configmodels.Type {
 	return TypeStr
 }
 
 // CreateDefaultConfig returns default configuration of Factory.
-// This function implements OTEL exporter.BaseFactory interface.
+// This function implements OTEL exporter.ExporterFactoryBase interface.
```
Comment (objectiser) [inline cmd/opentelemetry-collector/app/exporter/kafka/factory.go:53]: I think this should be `component.ExporterFactoryBase`?

**Comment by pavolloffay** [comment]
`go` service is failing in `CROSSDOCK_OTEL` I am not sure why. It is passing in normal crossdock.

File: cmd/opentelemetry-collector/app/exporter/cassandra/factory.go [hunk 2]
```diff
@@ -62,17 +64,25 @@ func (f Factory) CreateDefaultConfig() configmodels.Exporter {
 }
 
 // CreateTraceExporter creates Jaeger Cassandra trace exporter.
-// This function implements OTEL component.Factory interface.
-func (Factory) CreateTraceExporter(log *zap.Logger, cfg configmodels.Exporter) (component.TraceExporterOld, error) {
+// This function implements OTEL component.ExporterFactory interface.
+func (f Factory) CreateTraceExporter(
+	_ context.Context,
```
Comment (yurishkuro) [inline cmd/opentelemetry-collector/app/exporter/cassandra/factory.go:69]: why do they pass context? Is it used as a replacement for `Close()`?

Comment (pavolloffay) [reply cmd/opentelemetry-collector/app/exporter/cassandra/factory.go:69]: Close has its own context. This context seems to be associated with application start event.

**Comment by pavolloffay** [comment]
> I thought a processor had to be provided otherwise it would be an invalid OTC config?


This was the case before now It seems to work, I am trying to investigate xdock failure. The processors will be handled as part of https://github.com/jaegertracing/jaeger/issues/2152

**Comment by pavolloffay** [comment]
The processors were causing issues in the CI. Without the processors CI seems to pass fine. I will keep them removed and install back in #2152

**Pull request merged**